lib/timescaledb/schema_dumper.rb

```diff
@@ -4,16 +4,22 @@
 module Timescaledb
   module SchemaDumper
     def tables(stream)
+      # Create fake streams to capture the output for these objects
```

Comment:
are the instance variables useful only to capture the values in the tests?

Reply:
They're not actually used for the tests at all.

I use them to store the output from the `timescale_hypertable` and `timescale_retention_policy` methods when we call them as part of the `table` method; I'm then adding them to the end of the main stream at the end of the process.

This saves iterating over all of the tables multiple times, but I'm not 100% sold on the approach.

Reply:
yeah, me either. I don't get it in the first moment and the previous way was just easy to read. as this is also a process that runs once in a while, we could cache the tables in memory in case it becomes very slow.

Reply:
In that case, I'll switch my approach around as discussed in https://github.com/jonatas/timescaledb/pull/34#discussion_r1038154372 - this should make it more obvious.